onnxscript/tests/models/getitem.py

```diff
@@ -3,115 +3,183 @@
 # Licensed under the MIT License.
 # --------------------------------------------------------------------------
 
+from __future__ import annotations
+
+from typing import Union
+
+import numpy as np
 from onnx import TensorProto
 from onnx.helper import make_tensor
 
 from onnxscript import script
 from onnxscript.onnx_opset import opset15 as op
-from onnxscript.onnx_types import FLOAT
+from onnxscript.onnx_types import INT32, INT64
+from onnxscript.tests.common.onnx_script_test_case import FunctionTestParams
+
+x = np.array([[0, 1, 2], [3, 4, 5], [6, 7, 8], [9, 10, 11]], dtype=np.int32)
+zero = np.array(0, dtype=np.int64)
+
+# Inputs/Outputs of test-cases are specified as numpy arrays, or tuples of numpy arrays,
+# or as a list of values, e.g. [0, 1, 2], converted to an int32 numpy array.
+
+IOType = Union[np.ndarray, tuple[np.ndarray, ...], list]
+
+
+def wrap_input_output(x: IOType) -> list(np.ndarray):
+    if isinstance(x, np.ndarray):
+        return [x]
+    elif isinstance(x, tuple):
+        return list(x)
+    else:
+        return [np.array(x, dtype=np.int32)]
+
+
+def test(f, input: IOType, output: IOType) -> FunctionTestParams:
+    return FunctionTestParams(f, wrap_input_output(input), wrap_input_output(output))
 
 
 @script(default_opset=op)
-def getitem_rev0(A: FLOAT[...]) -> FLOAT[...]:
-    return A[0, :0:-1]
+def get_first_row(A: INT32[...]) -> INT32[...]:
+    return A[0]
+
+
+get_first_row_test = test(get_first_row, input=x, output=[0, 1, 2])
 
 
 @script(default_opset=op)
-def getitem_rev(A: FLOAT[...]) -> FLOAT[...]:
-    return A[:0:-1]
+def get_last_row(A: INT32[...]) -> INT32[...]:
+    return A[-1]
 
 
-@script()
-def getitem_index_int0(A: FLOAT[...]) -> FLOAT[...]:
-    zero = op.Constant(value=make_tensor("zero", TensorProto.INT64, [1], [0]))
-    r = A[zero]
-    return r
+get_last_row_test = test(get_last_row, input=x, output=[9, 10, 11])
```

Comment:
Possible to parameterize with `parameterized` instead?

Reply:
How? Why? I would like to keep the script-function being tested, and the input/output values together in one place (makes it easier for the reader). The existing structure achieves this.

Reply:
I think the follow can achieve this (keeping things together) while making the test case look like the standard tests:

```python
import parameterized


class GetItemTest(unittest.TestCase):
    @parameterized.parameterized.expand([
        ("get_first_row", get_first_row, x, [0, 1, 2]),
        ("get_last_row", get_last_row, x, [9, 10, 11]),
    ])
    def test_function(self, _: str, function, input, output):
        FunctionTestParams(function, wrap_input_output(input), wrap_input_output(output))

```

reference: https://pypi.org/project/parameterized/

Reply:
I understand the use of parameterized. But it still will split the definition of `get_first_row` from the test-case that tests it ... e.g., when you have 15 such functions. Your suggestion would apply to the existing test-case definition [here](https://github.com/microsoft/onnxscript/blob/bcee0ecf57506b8e6cae5dc7c93ae5195ee7071b/onnxscript/converter_test.py#L435) ... but I found it tedious to correlate these test-cases to the definition of the function being tested. (Of course, the situation was worsened since the definitions were in a different file.)

Maybe we can define a function that combines `@script` and `@parametrized` into one decorator. Something for a future PR.